copilot/__init__.py

```diff
@@ -0,0 +1,116 @@
+import ast
+import sys
+import http.client
+import json
+from importlib._bootstrap import spec_from_loader
+
+
+def get_suggestion(file, snippet, stops):
+    conn = http.client.HTTPSConnection("copilot.githubassets.com")
+    payload = json.dumps({
+        "prompt": "// Language: python\n// Path: " + file + "\n" + snippet,
+        "max_tokens": 200,
+        "temperature": 0.2,
+        "top_p": 1,
+        "n": 1,
+        "logprobs": 2,
+        "stop": stops,
+        "stream": False
+    })
+    headers = {
+        'OpenAI-Intent': 'copilot-ghost',
+        'OpenAI-Organization': 'github-copilot',
+        'Authorization': 'Bearer TOKEN',
+        'Content-Type': 'application/json'
+    }
+    conn.request("POST", "/v1/engines/github-py-stochbpe-cushman-pii/completions", payload, headers)
+    res = conn.getresponse()
+    data = res.read()
+    return json.loads(data.decode("utf-8"))
+
+
+def get_fn(name):
+    args = get_suggestion(name + ".py", "def " + name, ["):\n"])["choices"][0]["text"]
+    full_snippet = "def " + name + args + "):\n"
+
+    code = get_suggestion(name + ".py", full_snippet, [
+        "\n\n\n",
+        "\ndef ",
+        "\nif "
+    ])["choices"][0]["text"]
+
+    full_code = full_snippet + code
+
+    try:
+        block_ast = ast.parse(full_code)
+
+        if len(block_ast.body) == 1:
+            return full_code
+        else:
+            return "\n".join(full_code.split("\n")[:block_ast.body[1].lineno - 1])
```

Comment:
what's the logic here?

Reply:
in case copilot returns more than a single function, should ditch the extra code (or so I'm told)